src/lib/util.js

```diff
@@ -751,3 +751,61 @@ export function pageNumberFromScroll(currentPageNum, previousScrollTop, currentP
 
     return pageNum;
 }
+
+/**
+ * Calculates the midpoint from two points.
+ *
+ * @public
+ * @param {number} x1 - The x value of the first point
+ * @param {number} y1 - The y value of the first point
+ * @param {number} x2 - The x value of the second point
+ * @param {number} y2 - The y value of the second point
+ * @return {Array} The resulting x,y point
+ */
+export function getMidpoint(x1, y1, x2, y2) {
+    const x = (x1 + x2) / 2;
+    const y = (y1 + y2) / 2;
+    return [x, y];
+}
+
+/**
+ * Calculates the distance between two points.
+ *
+ * @public
+ * @param {number} x1 - The x value of the first point
+ * @param {number} y1 - The y value of the first point
+ * @param {number} x2 - The x value of the second point
+ * @param {number} y2 - The y value of the second point
+ * @return {number} The resulting distance
+ */
+export function getDistance(x1, y1, x2, y2) {
+    return Math.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2);
+}
+
+/**
+ * Returns the closest visible page to a pinch event
+ *
+ * @public
+ * @param {number} x - The x value of pinch
+ * @param {number} y - The y value of the pinch
+ * @param {Object} visiblePages - Object of visible pages to consider
+ * @return {HTMLElement} The resulting page
+ */
+export function getClosestPageToPinch(x, y, visiblePages) {
+    let closestPage = null;
+    for (let i = visiblePages.first.id; i <= visiblePages.last.id; i++) {
+        const page = document.querySelector(`#bp-page-${i}`);
+        const pageMidpoint = getMidpoint(
+            page.offsetLeft,
+            page.offsetTop,
+            page.offsetLeft + page.scrollWidth,
+            page.offsetTop + page.scrollHeight
+        );
+        const distance = getDistance(pageMidpoint[0], pageMidpoint[1], x, y);
+        if (!closestPage || distance < closestPage[1]) {
+            closestPage = [page, distance];
+        }
+    }
+
+    return closestPage[0];
```

Comment:
Careful, this'll error out if `closestPage` is never set

Reply:
check line 795!